helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1258,4 +1292,57 @@ private void initPipeline(Thread eventThread, BaseControllerDataProvider cache)
     eventThread.setDaemon(true);
     eventThread.start();
   }
+
+  /**
+   * A wrapper class for the stateful rebalancer instance that will be tracked in the
+   * GenericHelixController.
+   */
+  private abstract class StatefulRebalancerRef<T extends StatefulRebalancer> {
```

Comment:
1. Should we think about which methods in this class should be idempotent?

2. Why an abstract class vs an interface?

Reply:
Since it is a private class, I don't think we need to overdesign it. As for idempotent, could you be more specific, please? What's the concern that you have?

Reply:
My concern is that we should probably make the behavior more explicit around create() or close() being multiple times. It wasn't very clear from the code?

Reply:
My point is that we don't restrict the behavior in the Ref class. The source of truth is in the rebalancer object itself. So it won't be explicitly mentioned here.
Moreover, as I commented in the other thread, it is not a real concern for now. But I have added a TODO in the rebalancer class for rejecting requests after close() called.